src/app.rs

```diff
@@ -685,6 +685,19 @@ pub fn bye(code: i32, msg: &str) -> ! {
     std::process::exit(code);
 }
 
+macro_rules! bye_with_sdk_error {
+    ($code:expr,$error:expr) => {
+        use log::error;
+
+        match $error.as_service_error() {
+            Some(service_error) => error!("service error occurred: {:?}", service_error.meta()),
+            None => error!("an error occurred: {:?}", $error),
+        };
+        std::process::exit($code);
+    };
+}
+pub(crate) use bye_with_sdk_error;
```

Comment:
While we can define a regular function for this case, are there any specific reasons to use macros instead? I'm not familiar with the rationale behind choosing macros over a regular function.

```suggestion
pub fn bye_with_sdk_error<E: fmt::Debug + ProvideErrorMetadata, R: fmt::Debug>(code: i32, error: SdkError<E, R>) -> ! {
    use log::error;

    match error.as_service_error() {
        Some(service_error) => error!("service error occurred: {:?}", service_error.meta()),
        None => error!("an error occurred: {:?}", error),
    };
    std::process::exit(code);
}
```

Reply:
Perviously I misunderstand we need to use macros instead of function because I missed ProvideErrorMetadata trait. So we can use function pattern actually.